cargo-pgx/src/command/install.rs

```diff
@@ -459,7 +476,17 @@ pub(crate) fn format_display_path(path: impl AsRef<Path>) -> eyre::Result<String
     Ok(out)
 }
 
-fn filter_contents(manifest_path: impl AsRef<Path>, mut input: String) -> eyre::Result<String> {
+fn filter_contents(
+    manifest_path: impl AsRef<Path>,
+    mut input: String,
+    versioned_so: bool,
+) -> eyre::Result<String> {
+    if versioned_so {
+        input = input.replace(
+            "module_pathname =",
+            "# commented out by pgx because versioned .so requested\n# module_pathname =",
+        )
```

Comment:
I'm having a hard time visualizing what this is replacing, and where.  Can you clarify?

Reply:
In the `.control` file, one can set the `module_pathname` (path to shared object). With the default pgx extension template, the value of `module_pathname` is set to `$libdir/<extension-name>`. When using the `--versioned-so` feature, this default value for `module_pathname` will be incorrect.

Instead if leaving it in there to break stuff, this change comments it out of the `.control` file during the install phase.

Another option would be to check if a `module_pathname` entry is present in the `.control` file when `--versioned-so` is used, and explicitly throw an error. Actually I think that this approach may be more explicit in what it does.

Reply:
Another other option would be to make `module_pathname` templated in the control file, for instance setting it to `@MODULE_PATHNAME@`. This would bring it in line with how `default_version` works.

Reply:
Based on Postgres' docs for `module_pathname`, it really doesn't mean anything unless `MODULE_PATHNAME` exists in any of the extension sql scripts.  And with this PR, I guess we wouldn't.

Here's an alternative thought...

What if instead of having the `--versioned-so` CLI arg, we instead look for `module_pathname` in the .control file.  If it's there, we do the default behavior, if it's not, then we do the versioned-so behavior?  

I'm still not 100% convinced that if you want versioned .so's that you'd want to turn them off from time to time or from cargo pgx command to command.

Thoughts?

Reply:
> And with this PR, I guess we wouldn't.

Yes and no. By user error, it's possible that `MODULE_PATHNAME` would still be set in an SQL override.

> What if instead of having the --versioned-so CLI arg, we instead look for module_pathname in the .control file. If it's there, we do the default behavior, if it's not, then we do the versioned-so behavior?

That's an elegant way to "hide" the optionality. I kinda like it. I can't think of a scenario which it causes problems with, but maybe that's my lack of creativity. One downside it that it's quite implicit.

> I'm still not 100% convinced that if you want a versioned .so that there'd also be times when you wouldn't. That doesn't jive in my little brain.

I think that you're right on this.

Reply:
> You wanna go ahead and make these changes? Sorry for the churn, but we really need to get this right so it's not surprising to users.

Yup, I've changed it to work as you described. Please take another look.